parser.y

```diff
@@ -6324,84 +6327,105 @@ TableOptimizerHintList:
+|	hintUseToja '(' QueryBlockOpt HintTrueOrFalse ')'
 	{
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), HintFlag: $3.(bool)}
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QBName: $3.(model.CIStr), HintFlag: $4.(bool)}
 	}
-|	hintEnablePlanCache '(' HintTrueOrFalse ')'
+|	hintEnablePlanCache '(' QueryBlockOpt HintTrueOrFalse ')'
 	{
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), HintFlag: $3.(bool)}
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QBName: $3.(model.CIStr), HintFlag: $4.(bool)}
 	}
-|	maxExecutionTime '(' NUM ')'
+|	maxExecutionTime '(' QueryBlockOpt NUM ')'
 	{
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), MaxExecutionTime: getUint64FromNUM($3)}
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QBName: $3.(model.CIStr), MaxExecutionTime: getUint64FromNUM($4)}
 	}
-|	hintUsePlanCache '(' ')'
+|	hintUsePlanCache '(' QueryBlockOpt ')'
 	{
 		// arguments not decided yet.
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1)}
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QBName: $3.(model.CIStr)}
 	}
-|	hintQueryType '(' HintQueryType ')'
+|	hintQueryType '(' QueryBlockOpt HintQueryType ')'
 	{
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QueryType: model.NewCIStr($3.(string))}
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QBName: $3.(model.CIStr), QueryType: model.NewCIStr($4.(string))}
 	}
-|	hintMemoryQuota '(' HintMemoryQuota ')'
+|	hintMemoryQuota '(' QueryBlockOpt HintMemoryQuota ')'
 	{
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), MemoryQuota: $3.(uint64)}
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QBName: $3.(model.CIStr), MemoryQuota: $4.(uint64)}
 	}
-|	hintHASHAGG
+|	hintHASHAGG '(' QueryBlockOpt ')'
 	{
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1)}
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QBName: $3.(model.CIStr)}
 	}
-|	hintSTREAMAGG
+|	hintSTREAMAGG '(' QueryBlockOpt ')'
 	{
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1)}
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QBName: $3.(model.CIStr)}
 	}
-|	hintNoIndexMerge
+|	hintNoIndexMerge '(' QueryBlockOpt ')'
 	{
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1)}
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QBName: $3.(model.CIStr)}
 	}
-|	hintReadConsistentReplica
+|	hintReadConsistentReplica '(' QueryBlockOpt ')'
 	{
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1)}
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QBName: $3.(model.CIStr)}
+	}
+|	hintQBName '(' Identifier ')'
+	{
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QBName: model.NewCIStr($3)}
 	}
 
-HintTableAndIndexList:
-	Identifier
+QueryBlockOpt:
 	{
-		$$ = []model.CIStr{model.NewCIStr($1)}
+		$$ = model.NewCIStr("")
 	}
-|	HintTableAndIndexList ',' Identifier
+|	singleAtIdentifier
```

Comment:
I'm wondering whether we can use comma to split arguments in SQL hints too, it looks better.
```suggestion
|	singleAtIdentifier
	{
		$$ = model.NewCIStr($1)
	}
|	singleAtIdentifier ‘,’
```

Reply:
No, even if you want `,`, it should not be put here.